cmd/security-secrets-setup/testdata/test-example-security.yml

```diff
@@ -0,0 +1,625 @@
+#  * or implied. See the License for the specific language governing permissions and limitations under
+#  * the License.
+#  *
+#  * @author: Jim White, Dell
+#  * added: July 26, 2019
+#  * @author: Jim Wang, Intel Corp
+#  * updated: October 8, 2019
+#  *******************************************************************************/
+
+version: '3.4'
+
+# all common shared environment variables defined here:
+x-common-env-variables: &common-variables
+  edgex_registry: consul://edgex-core-consul:8500
+  Clients_CoreData_Host: edgex-core-data
+  Clients_Logging_Host: edgex-support-logging
+  Logging_EnableRemote: "true"
+
+volumes:
+  db-data:
+  log-data:
+  consul-config:
+  consul-data:
+  consul-scripts:
+  portainer_data:
+  vault-init:
+  vault-config:
+  vault-file:
+  vault-logs:
+  secrets-setup-cache:
+
+services:
+  volume:
+    image: nexus3.edgexfoundry.org:10004/docker-edgex-volume:1.1.0
+    container_name: edgex-files
+    networks:
+      - edgex-network
+    volumes:
+      - db-data:/data/db
+      - log-data:/edgex/logs
+      - consul-config:/consul/config
+      - consul-data:/consul/data
+
+  consul:
+    #image: nexus3.edgexfoundry.org:10004/docker-edgex-consul:1.1.0
+    image: edgexfoundry/docker-edgex-consul:1.1.1
+    ports:
+      - "8400:8400"
+      - "8500:8500"
+    container_name: edgex-core-consul
+    hostname: edgex-core-consul
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-core-consul
+    volumes:
+      - db-data:/data/db
+      - log-data:/edgex/logs
+      - consul-config:/consul/config
+      - consul-data:/consul/data
+      - consul-scripts:/consul/scripts
+      - vault-config:/vault/config
+      - /run/edgex/secrets:/run/edgex/secrets:ro    
+    depends_on:
+      - volume
+
+  config-seed:
+    image: nexus3.edgexfoundry.org:10004/docker-core-config-seed-go:1.1.0
+    container_name: edgex-config-seed
+    hostname: edgex-core-config-seed
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-core-config-seed
+    volumes:
+      - db-data:/data/db
+      - log-data:/edgex/logs
+      - consul-config:/consul/config
+      - consul-data:/consul/data
+    depends_on:
+      - volume
+      - consul
+
+  vault:
+    image: vault:1.0.3
+    container_name: edgex-vault
+    hostname: edgex-vault
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-vault
+    ports:
+      - "8200:8200"
+    cap_add:
+      - "IPC_LOCK"
+    entrypoint: ["/vault/init/start_vault.sh"]
+    environment:
+      - VAULT_ADDR=https://edgex-vault:8200
+      - VAULT_LOCAL_CONFIG=${VAULT_LOCAL_CONFIG}
+      - VAULT_CONFIG_DIR=/vault/config
+      - VAULT_UI=true
+    volumes:
+      - vault-file:/vault/file
+      - vault-logs:/vault/logs
+      - vault-init:/vault/init:ro
+      - /run/edgex/secrets/edgex-vault:/run/edgex/secrets/edgex-vault:ro
+    depends_on:
```

Comment:
Suggest lock down this container some more by adding
```
read_only: true
tmpfs:
  - /vault/config
```

Reply:
sounds good, I'll try that

Reply:
cannot set `read_only`: it is conflicting with `setcap` from /bin/vault.

Reply:
Technically, it is docker-entrypoint.sh that is doing the setcap.  Understood.